tests/testAutopas/tests/containers/AllContainersTests.cpp

```diff
@@ -0,0 +1,149 @@
+/**
+ * @file TestsAllContainers.cpp
+ * @author humig
+ * @date 08.07.2019
+ */
+
+#include "AllContainersTests.h"
+
+#include <gtest/gtest.h>
+
+#include "autopas/utils/StringUtils.h"
+
+using ::testing::_;
+using ::testing::AtLeast;
+using ::testing::Each;
+using ::testing::Eq;
+using ::testing::Invoke;
+using ::testing::Values;
+
+INSTANTIATE_TEST_SUITE_P(Generated, AllContainersTests, testing::ValuesIn([]() {
+                           auto allOptions = std::set<autopas::ContainerOption>{autopas::ContainerOption::getAllOptions()};
+                           allOptions.erase(autopas::ContainerOption::verletClusterLists);
+                           return allOptions;
+                         }()),
+                         AllContainersTests::getParamToStringFunction());
+
+INSTANTIATE_TEST_SUITE_P(DISABLED_Generated, AllContainersTests,
+                         testing::Values(autopas::ContainerOption::verletClusterLists),
+                         AllContainersTests::getParamToStringFunction());
+
+/**
+ * Checks if ParticleContainer::getNumParticle() returns the correct number of particles.
+ *
+ * @Reviewer: Meant to replace
+ *      VerletListsTest::testAddParticleNumParticle
+ *      LinkedCellsTest::testGetNumParticles
+ *      DirectSumContainerTest::testGetNumParticles
+ */
+TEST_P(AllContainersTests, testGetNumParticles) {
+  EXPECT_EQ(_container->getNumParticles(), 0);
+
+  std::array<double, 3> r = {2, 2, 2};
+  Particle p(r, {0., 0., 0.}, 0);
+  _container->addParticle(p);
+  EXPECT_EQ(_container->getNumParticles(), 1);
+
+  std::array<double, 3> r2 = {1.5, 2, 2};
+  Particle p2(r2, {0., 0., 0.}, 1);
+  _container->addParticle(p2);
+  EXPECT_EQ(_container->getNumParticles(), 2);
+}
+
+/**
+ * Checks if ParticleContainer::deleteAllParticles() deletes all particles.
+ *
+ * @Reviewer: Meant to replace
+ * 	VerletListsTest::testDeleteAllParticles
+ *      LinkedCellsTest::testDeleteAllParticles
+ *      DirectSumContainerTest::testDeleteAllParticles
+ */
+TEST_P(AllContainersTests, testDeleteAllParticles) {
+  EXPECT_EQ(_container->getNumParticles(), 0);
+
+  std::array<double, 3> r = {2, 2, 2};
+  Particle p(r, {0., 0., 0.}, 0);
+  _container->addParticle(p);
+
+  std::array<double, 3> r2 = {1.5, 2, 2};
+  Particle p2(r2, {0., 0., 0.}, 1);
+  _container->addParticle(p2);
+  EXPECT_EQ(_container->getNumParticles(), 2);
+
+  _container->deleteAllParticles();
+  EXPECT_EQ(_container->getNumParticles(), 0);
+}
+
+/**
+ * Checks if addParticle() only accepts particles in the domain and throws otherwise, and if addHaloParticle() never
+ * throws.
+ *
+ * @Reviewer: Meant to replace
+ *      LinkedCellsTest::testParticleAdding
+ *      DirectSumContainerTest::testParticleAdding
+ */
+TEST_P(AllContainersTests, testParticleAdding) {
```

Comment:
this test partially fails in the moment for some tests.
it's also no longer correct (see below)